### PR TITLE
[PLATFORM-1917] Added delete local group maintenance script

### PR DESCRIPTION
**Pull request by jcellary**
Added a script which can delete local groups - for starters supports only upwizcampeditors, but can be easily expanded.

@wladekb

### CODE REVIEW

File: maintenance/wikia/deleteLocalGroup.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			$this->output( "Deleting group on {$wgDBname} ({$wgDBCluster})\n" );
+			$dbw = $this->getDB( DB_MASTER );
+			$dbw->begin();
+			$dbw->delete(
+				self::TABLE_NAME,
+				[ 'ug_group' => [ $group ] ],
+				__METHOD__
+			);
+			$dbw->commit();
+			wfWaitForSlaves();
```
Comment (wladekb) [inline maintenance/wikia/deleteLocalGroup.php]: Any reason for that?

File: maintenance/wikia/deleteLocalGroup.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		}
+
+		$then = microtime( true );
+		if ( !$delete ) {
+			$this->output( "Getting entries count for group on {$wgDBname} ({$wgDBCluster})\n" );
+			$dbw = $this->getDB( DB_SLAVE );
+
+			$res = $dbw->select(
+				self::TABLE_NAME,
+				[ 'ug_user', 'ug_group' ],
```
Comment (wladekb) [inline maintenance/wikia/deleteLocalGroup.php]: Any reason you're not pulling COUNT(*) from db here?

**Comment by wladekb** [comment]
left 2 comments

File: maintenance/wikia/deleteLocalGroup.php (hunk 1 above)
Comment (jcellary) [inline maintenance/wikia/deleteLocalGroup.php]: No

File: maintenance/wikia/deleteLocalGroup.php (hunk 2 above)
Comment (jcellary) [inline maintenance/wikia/deleteLocalGroup.php]: No

**Comment by jcellary** [comment]
@wladekb please check again

File: maintenance/wikia/deleteLocalGroup.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		if ( !in_array( $group, self::$allowed_groups ) ) {
+			die( "Error: requested group " . $group . " not in the allowed white-list\n" );
+		}
+
+		$then = microtime( true );
+		if ( !$delete ) {
+			$this->output( "Getting entries count for group on {$wgDBname} ({$wgDBCluster})\n" );
+			$dbw = $this->getDB( DB_SLAVE );
+
+			$res = $dbw->select(
```
Comment (wladekb) [inline maintenance/wikia/deleteLocalGroup.php]: use `selectField` which returns a single scalar value so you don't need the weird expression:
`->fetchRow()[0]` a few lines below

File: maintenance/wikia/deleteLocalGroup.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+			);
+
+			$this->output( "Row count: " . $res->fetchRow()[0] . "\n" );
+		} else {
+			$this->output( "Deleting group on {$wgDBname} ({$wgDBCluster})\n" );
+			$dbw = $this->getDB( DB_MASTER );
+			$dbw->begin();
+			$dbw->delete(
+				self::TABLE_NAME,
+				[ 'ug_group' => [ $group ] ],
```
Comment (wladekb) [inline maintenance/wikia/deleteLocalGroup.php]: I guess you want just:
`[ 'ug_group' => $group ],`

File: maintenance/wikia/deleteLocalGroup.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+		$then = microtime( true );
+		if ( !$delete ) {
+			$this->output( "Getting entries count for group on {$wgDBname} ({$wgDBCluster})\n" );
+			$dbw = $this->getDB( DB_SLAVE );
+
+			$res = $dbw->select(
+				self::TABLE_NAME,
+				'count(*)',
+				[ 'ug_group' => [ $group ] ],
```
Comment (wladekb) [inline maintenance/wikia/deleteLocalGroup.php]: I guess you want just:
`[ 'ug_group' => $group ],`

**Comment by jcellary** [comment]
Thank you @wladekb I included suggested fixes, please let me know if there is anything else.

**Comment by wladekb** [comment]
lgtm

**Pull request merged**